float64[] middle_line
float64 middle_curvatures
float64[] worldmiddle_point